pkg/controller/cloud/node_controller.go

```diff
@@ -42,6 +43,35 @@ import (
 	nodeutil "k8s.io/kubernetes/pkg/util/node"
 )
 
+// labelReconcileInfo lists Node labels to reconcile, and how to reconcile them.
+// primaryKey and secondaryKey are keys of labels to reconcile.
+//   - If both keys exist, but their values don't match. Use the value from the
+//   primaryKey as the source of truth to reconcile.
+//   - If ensureSecondaryExists is true, and the secondaryKey does not
+//   exist, secondaryKey will be added with the value of the primaryKey.
+var labelReconcileInfo = []struct {
```

Comment:
this mimics the os/arch label controller code in https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L136-L165

would be nice if it could be deduplicated, but I won't block on that